cloud/amazon/ec2_eni.py

```diff
@@ -466,7 +516,7 @@ def main():
                             ],
                            required_if=([
                                ('state', 'present', ['subnet_id']),
-                               ('state', 'absent', ['eni_id']),
+                               ('state', 'absent', ['eni_id'])
```

Comment:
Attached defaults to true, and may be specified without instance_id in the case where the user doesn't want to bother with attachment, only the ENI existence. In that case, attachment_status of the interface wouldn't change.